validation/unique.go

```diff
@@ -144,6 +148,68 @@ func (v *ExistsArrayValidator[T]) buildQuery(values []T, condition bool) *gorm.D
 	return db.Raw(sql, params...)
 }
 
+// Build Query for Clickhouse syntax
+func (v *ExistsArrayValidator[T]) buildClickhouseQuery(values []T, condition bool) *gorm.DB {
+	questionMarks := []string{}
+	params := []any{}
+
+	paramZeroType := ""
+	paramOneType := ""
+
+	for i, val := range values {
+		questionMarks = append(questionMarks, "?")
+		var transformedValue any = val
+		if v.Transform != nil {
+			transformedValue = v.Transform(val)
+		}
+		params = append(params, gorm.Expr(
+			"(?,?)",
+			transformedValue, i))
+		if paramZeroType == "" {
+			paramZeroType = clickhouseTypes[reflect.TypeOf(transformedValue)]
+		}
+		if paramOneType == "" {
+			paramOneType = clickhouseTypes[reflect.TypeOf(i)]
+		}
+	}
+
+	db := v.DB()
+	table := db.Statement.Quote(v.Table)
+	column := db.Statement.Quote(v.Column)
+
+	sql := fmt.Sprintf(
+		"WITH ctx_values(id, i) AS (SELECT * FROM (VALUES 'id %s, i %s', %s)) SELECT i FROM ctx_values INNER JOIN %s ON %s.%s = ctx_values.id WHERE %s.%s IS %s NULL",
+		paramZeroType,
+		paramOneType,
+		strings.Join(questionMarks, ","),
+		table,
+		table, column,
+		table, column,
+		lo.Ternary(condition, "", "NOT"),
+	)
+	return db.Raw(sql, params...)
+}
+
+// Mapping of Go types to Clickhouse types can be found here:
+// https://github.com/ClickHouse/clickhouse-go/blob/main/TYPES.md
+// Go types uint and int are not specified, default to 'UInt64' and 'Int64', respectively
+var clickhouseTypes = map[reflect.Type]string{
+	reflect.TypeOf(uint64(0)):  "UInt64",
+	reflect.TypeOf(uint32(0)):  "UInt32",
+	reflect.TypeOf(uint16(0)):  "UInt16",
+	reflect.TypeOf(uint8(0)):   "UInt8",
+	reflect.TypeOf(uint(0)):    "UInt64",
+	reflect.TypeOf(int64(0)):   "Int64",
+	reflect.TypeOf(int32(0)):   "Int32",
+	reflect.TypeOf(int16(0)):   "Int16",
+	reflect.TypeOf(int8(0)):    "Int8",
+	reflect.TypeOf(int(0)):     "Int64",
+	reflect.TypeOf(float32(0)): "Float32",
+	reflect.TypeOf(float64(0)): "Float64",
+	reflect.TypeOf(""):         "String",
+	reflect.TypeOf(true):       "Bool",
+}
```

Comment:
Nice job, that will do. This way we don't have to import an additional package.
I am just worried that it won't work with types like uuid, time, and the other type-conversion validators. Let's add the most common ones and let users handling the custom ones with the transform function.